File: 加密保护/VMProtect Professional v2.08/Examples/Delphi/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls;

type
  TForm1 = class(TForm)
    Label1: TLabel;
    Edit1: TEdit;
    Button1: TButton;
    procedure Button1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

uses 
  VMProtectSDK;

{$R *.DFM}

procedure TForm1.Button1Click(Sender: TObject);
begin
 VMProtectBegin('Test marker');
 if StrToIntDef(Edit1.Text, 0) mod 17=13 then
  MessageDlg(VMProtectDecryptStringA('Correct password'), mtInformation, [mbOK], 0)
 else
  begin
   MessageDlg(VMProtectDecryptStringA('Incorrect password'), mtError, [mbOK], 0);
   Edit1.SetFocus;
  end;
 VMProtectEnd;
end;

end.

File: 加密保护/VMProtect Professional v2.08/Examples/Borland C++ Builder/Unit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "dialogs.hpp"
#include "Unit1.h"
#include "VMProtectSDK.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::Button1Click(TObject *Sender)
{
 VMProtectBegin("Test marker");
 if (StrToIntDef(Edit1->Text, 0) % 17==13)
  MessageDlg(VMProtectDecryptStringA("Correct password"), mtInformation, TMsgDlgButtons() << mbOK, 0);
 else
  {
   MessageDlg(VMProtectDecryptStringA("Incorrect password"), mtError, TMsgDlgButtons() << mbOK, 0);
   Edit1->SetFocus();
  }
 VMProtectEnd();
}
//---------------------------------------------------------------------------
